ostd/src/mm/page_table/boot_pt.rs

```diff
@@ -160,33 +162,37 @@ impl<E: PageTableEntryTrait, C: PagingConstsTrait> BootPageTable<E, C> {
     ///
     /// This function is unsafe because it can cause undefined behavior if the caller
     /// maps a page in the kernel address space.
-    pub unsafe fn map_base_page(&mut self, from: Vaddr, to: FrameNumber, prop: PageProperty) {
+    pub unsafe fn map_base_page(&mut self, from: Vaddr, to: Paddr, prop: PageProperty) {
         let mut pt = self.root_pt;
         let mut level = C::NR_LEVELS;
         // Walk to the last level of the page table.
         while level > 1 {
             let index = pte_index::<C>(from, level);
-            let pte_ptr = unsafe { (paddr_to_vaddr(pt * C::BASE_PAGE_SIZE) as *mut E).add(index) };
+            let pte_ptr = unsafe { (paddr_to_vaddr(pt) as *mut E).add(index) };
             let pte = unsafe { pte_ptr.read() };
-            pt = if !pte.is_present() {
-                let pte = self.alloc_child();
-                unsafe { pte_ptr.write(pte) };
-                pte.paddr() / C::BASE_PAGE_SIZE
-            } else if pte.is_last(level) {
-                panic!("mapping an already mapped huge page in the boot page table");
-            } else {
-                pte.paddr() / C::BASE_PAGE_SIZE
+            match pte.to_repr(level) {
+                PteScalar::Absent => {
+                    let (pte, ch_pt_pa) = self.alloc_child(level);
+                    unsafe { pte_ptr.write(pte) };
+                    pt = ch_pt_pa;
+                }
+                PteScalar::Mapped(_, _) => {
+                    panic!("mapping an already mapped huge page in the boot page table");
+                }
+                PteScalar::PageTable(ch_pt_pa, _) => {
+                    pt = ch_pt_pa;
```

Comment:
`ch_pt_pa` -> `child_pt`?

It seems that only this place uses the prefix `ch_` (which is a bit odd) and that other places tend to use something like `child_pt` or `child_pt_addr`.